turbopack/crates/turbo-rcstr/src/dynamic.rs

```diff
@@ -0,0 +1,76 @@
+use std::{
+    borrow::Cow,
+    hash::{Hash, Hasher},
+    ptr::NonNull,
+};
+
+use triomphe::Arc;
+
+use crate::{
+    tagged_value::{TaggedValue, MAX_INLINE_LEN},
+    RcStr, INLINE_TAG_INIT, LEN_OFFSET, TAG_MASK,
+};
+
+#[derive(Debug)]
+pub(crate) struct Entry {
+    pub string: Arc<str>,
+}
```

Comment:
"Entry" makes sense in `hstr` because these are entries into a string intern cache, but I doubt we'll ever want to do interning with `RcStr` because turbo-tasks is already effectively doing some level of interning with it's cache.